### PR TITLE
feature: add `registerDestination`function

### PR DESCRIPTION
**Pull request by tomfrenken**
This PR adds the `registerDestination` function which sets a given destination into the environment variables.

Closes SAP/cloud-sdk-backlog#467.

### CODE REVIEW

File: packages/connectivity/src/scp-cf/destination/destination-from-env.ts [hunk 1]
```diff
@@ -168,3 +168,47 @@ function destinationAuthToken(
...
+
+/**
+ * Set a given destination in the `destinations` environment variable.
+ *
+ * Throws an error if a destination with the same name as the given test destination already exists.
+ * @param destination - Test destination to add to the `destinations` environment variable
+ */
+export function registerDestination(destination: Destination): void {
+  const currentDestinations = getDestinationsFromEnv();
+  const existingNames = new Set<string>(
```
Comment (FrankEssenberger) [inline packages/connectivity/src/scp-cf/destination/destination-from-env.ts:201]: we have a `unique` method in the util. This does the same but is more speaking.

File: packages/connectivity/src/scp-cf/destination/destination-from-env.spec.ts [hunk 2]
```diff
@@ -206,3 +207,27 @@ describe('env-destination-accessor', () => {
...
+    name: 'MockedDestination',
+    url: 'https://example.com'
+  };
+
+  afterEach(() => {
+    unmockDestinationsEnv();
+    jest.resetAllMocks();
+  });
+
+  it('should set the destination in the environment variables', () => {
```
Comment (FrankEssenberger) [inline packages/connectivity/src/scp-cf/destination/destination-from-env.spec.ts:222]: Add a test to see the errors are thrown for empty name destinations

Comment (FrankEssenberger) [reply packages/connectivity/src/scp-cf/destination/destination-from-env.spec.ts:222]: Also add a test for the `getDestination(name)` case to see that the destination by name is working.

Comment (FrankEssenberger) [reply packages/connectivity/src/scp-cf/destination/destination-from-env.spec.ts:222]: Also add a test to see the name conflict exception being raised.

File: packages/connectivity/src/scp-cf/destination/destination-from-env.ts [hunk 3]
```diff
@@ -168,3 +168,47 @@ function destinationAuthToken(
...
+    currentDestinations.map(dest => {
+      if (!dest.name) {
```
Comment (FrankEssenberger) [inline packages/connectivity/src/scp-cf/destination/destination-from-env.ts:203]: Why this is a problem? I think our set method should ensure that you can not set a destination with undefined name, but if they are in the env by some mechanism is this a problem? Also without some context you get an exception which you can not investigate. So perhaps use a warning and give some info on the destination. Be careful not to expose sensitve information in the log. So perhaps it is easier to just get rid of this exception?

File: packages/connectivity/src/scp-cf/destination/destination-from-env.ts [hunk 4]
```diff
@@ -168,3 +168,47 @@ function destinationAuthToken(
...
+        throw Error('The destination name is undefined.');
+      }
+      return dest.name;
+    })
+  );
+  if (!destination.name) {
+    throw Error('The destination name is undefined.');
```
Comment (FrankEssenberger) [inline packages/connectivity/src/scp-cf/destination/destination-from-env.ts:210]: Perhaps formulate with a clear action: `registerDestination` does require to set XYZ.

Comment (FrankEssenberger) [reply packages/connectivity/src/scp-cf/destination/destination-from-env.ts:210]: I would use a small helper method `checkDestiantionValid` which does a check on mandatory fields like name, but also URL.

**Review by FrankEssenberger — CHANGES_REQUESTED** [review]
Some request on more tests and exception handling.
